src/Umbraco.Web/Search/SearchableTreeCollectionBuilder.cs

```diff
@@ -21,5 +21,8 @@ public override SearchableTreeCollection CreateCollection()
         {
             return new SearchableTreeCollection(CreateItems(), _treeService);
         }
+
+        //per request because generally an instance of ISearchableTree is a controller
+        protected override ILifetime CollectionLifetime => new PerRequestLifeTime();
```

Comment:
Means that the collection will be re-created on each request instead of being cached as a singleton. Is this OK?

Reply:
As the comment says, ISearchableTree's are generally webapi controllers which must be per-request based. This used to be a singleton lifespan and that causes problems because then only 1 instance of any ISearchableTree is created which is bad because it takes a reference to a web-based UmbracoHelper which is stored and then exceptions occur. 

This collection should only get created on a request when the collection is needed, it shouldn't create the collection on every single request. 

We could refactor how it all works and move implementations of ISearchableTree's to other classes and not controller instances which means we could then make them singletons, however many of them will then require the IUmbracoContextAccessor

I think for now we leave this as-is and we can decide if we want to change the implementation after it is merged.

Reply:
So it's OK, fine.